database/migrations/0049-add_user_contacts_table.ts

```diff
@@ -41,47 +41,13 @@ export async function upgrade(queryFn: (query: string, values?: any[]) => Promis
 
   // merge values from login_email_opt_in table with users.email in new user_contacts table
   await queryFn(`
-  INSERT INTO user_contacts
-  (type, user_id, email, email_verification_code, email_opt_in_type_id, email_resend_count, email_checked, created_at, updated_at, deleted_at)
-  SELECT 
-    'EMAIL', 
-     u.id as user_id,
-     u.email,
-     e.verification_code as email_verification_code,
-     e.email_opt_in_type_id,
-     e.resend_count as email_resend_count,
-     u.email_checked,
-     e.created as created_at,
-     e.updated as updated_at,
-     u.deletedAt as deleted_at\
-   FROM 
-     users as u,
-     login_email_opt_in as e
-   WHERE 
-     u.id = e.user_id AND
-     e.id in (
-       WITH opt_in AS ( 
-         SELECT 
-           le.id, le.user_id, le.created, le.updated, ROW_NUMBER() OVER (PARTITION BY le.user_id ORDER BY le.created DESC) AS row_num
-         FROM 
-           login_email_opt_in as le
-       )
-       SELECT 
-         opt_in.id
-       FROM 
-         opt_in
-       WHERE
-         row_num = 1);`)
-  /*         
-      // SELECT 
-      //   le.id 
-      // FROM 
-      //   login_email_opt_in as le
-      // WHERE
-      //   le.user_id = u.id  
-      // ORDER BY 
-      //   le.updated DESC, le.created DESC LIMIT 1);`)
-      */
+    INSERT INTO \`user_contacts\`
+      (type, user_id, email, email_verification_code, email_opt_in_type_id, email_resend_count, email_checked, created_at, updated_at, deleted_at)
+      SELECT 'EMAIL', \`users\`.id, \`users\`.email, optin.\`verification_code\`, optin.\`email_opt_in_type_id\`, optin.\`resend_count\`, users.\`email_checked\`, users.created, null, users.deletedAt
+      FROM users LEFT JOIN
+      (SELECT le.id, le.user_id, le.verification_code, le.email_opt_in_type_id, le.resend_count, le.created, le.updated,
+         ROW_NUMBER() OVER (PARTITION BY le.user_id ORDER BY le.created DESC) AS row_num
+       FROM login_email_opt_in as le) AS optin ON users.id = optin.\`user_id\` AND row_num = 1;`)
```

Comment:
Ja das sieht logisch gut aus...
formell ist mir noch nicht klar warum du in Zeile 46 u 50 teilweise das users mit \` klammerst oder bei dem optin die attributnamen?
Kommt das weil in Zeile 46 das EMAIL in ` steht? Das ist aber auch ein konstanter Stringtext und keine Variable...

Ich kann daher schlecht einschätzen, ob das Statement mit den Klammerungen so funktioniert.
Ich hatte ja zeitgleich auf dem Branch #2225 das Statement ohne diese Klammern in die 0049er migration mal eingebaut und mit den produktiv Daten lokal getestet.

Reply:
@clauspeterhuebner I messed this up. When I tested the migration it did not work on the first run. I thought it had to do with the quotes, so this is a left over. I removed the back ticks from the sql statement.